test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -107,43 +101,40 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
                     }
                 }
 
-                #Write-Verbose -Verbose "Received $($results.Count) results in $($count * 300) milliseconds"
-
                 return $results
             }
 
-            # call this only within an It block
-            function checkContent($array)
+            function CheckContent($array)
             {
-                for ($i=1; $i -lt $array.Count; ++$i)
+                for ($i=1; $i -lt $array.Count; $i++)
                 {
-                    $array[$i] | should be ($array[$i-1] + 1)
+                    if ($array[$i] -ne ($array[$i-1] + 1))
+                    {
+                        return $false
+                    }
                 }
+
+                return $true
             }
 
         }
         BeforeEach {
-            $j = start-job -scriptblock { $count = 1; while ($true) { Write-Output ($count++); Start-Sleep -Milliseconds 30 } }
+            $j = Start-Job -ScriptBlock { $count = 1; while ($true) { Write-Output ($count++); Start-Sleep -Milliseconds 30 } }
         }
         AfterEach {
-            Get-Job | Stop-Job
-            Get-Job | Remove-Job -force
+            Get-Job | Remove-Job -Force
         }
 
         It "Receive-Job will retrieve partial output" {
-            $result1 = getResults $j 5 $false
-            $result2 = getResults $j 5 $false
-            checkContent ($result1 + $result2)
+            $result1 = GetResults $j 5 $false
+            $result2 = GetResults $j 5 $false
+            CheckContent ($result1 + $result2)
         }
         It "Receive-Job will retrieve partial output, including -Keep results" {
-            $result1 = getResults $j 5 $true
-            $result2 = getResults $j ($result1.Count + 5) $false
-            checkContent $result1
-            for ($i=0; $i -lt $result1.count; ++$i)
-            {
-                $result1[$i] | should be $result2[$i]
-            }
-            checkContent $result2
+            $result1 = GetResults $j 5 $true
+            $result2 = GetResults $j ($result1.Count + 5) $false
+            Compare-Object -PassThru $result1 $result2[0..($result1.Count-1)] | Should Be $null
```

Comment:
This is to compare the whole of `$result1` with the first _n_ elements of `$result2`, where _n_ is the length of `$result1`. Did the second `Receive-Job` return all of the data that the first one did?

Then it checks for any missing data by seeing if the element immediately following the first _n_ in `$result2` is the next number in sequence.